### PR TITLE
chore: update storybook to latest v8

### PR DESCRIPTION
**Pull request by aishu-nair**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: CDE-1847

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

- Add `autoprefixer` as a direct dependency because the Storybook update causes it not be hoisted.
- Refactor how full page screenshots are taken to remove the `scrollHeight` query. This query was flaky for the dropdown showcase story.
- Increase screenshot threshold for slight screenshot variances. No other solution has been found.

### CODE REVIEW

File: .storybook/stories/signpost/signpost.stories.ts [hunk 1]
```diff
@@ -56,18 +57,12 @@ const template = `
...
-export const Initial: StoryObj = {
-  render: SignpostTemplate,
-};
-
-export const Opened = {
-  render: SignpostTemplate,
-  async play({ canvasElement }) {
-    canvasElement.querySelector('button').click();
-  },
+export const Opened: StoryFn = Initial.bind({});
```
Comment (kevinbuhmann) [inline .storybook/stories/signpost/signpost.stories.ts:65]: This should be reverted.

File: .storybook/stories/spinner/spinner.stories.ts [hunk 2]
```diff
@@ -19,20 +19,24 @@ export default {
...
   argTypes: {
     // inputs
-    clrInline: { defaultValue: false, control: { type: 'boolean' } },
-    clrInverse: { defaultValue: false, control: { type: 'boolean' } },
-    clrMedium: { defaultValue: false, control: { type: 'boolean' } },
-    clrSmall: { defaultValue: false, control: { type: 'boolean' } },
+    clrInline: { control: { type: 'boolean' } },
+    clrInverse: { control: { type: 'boolean' } },
+    clrMedium: { control: { type: 'boolean' } },
+    clrSmall: { control: { type: 'boolean' } },
```
Comment (kevinbuhmann) [inline .storybook/stories/spinner/spinner.stories.ts:25]: These arg types are not needed. This affects many files.

File: .storybook/stories/stack-view/stack-view.stories.ts [hunk 3]
```diff
@@ -89,6 +99,35 @@ const StackViewAllTemplate: Story = args => ({
   props: args,
 });
 
+const StackViewAllTemplateWithPopOver: StoryFn = args => ({
```
Comment (kevinbuhmann) [inline .storybook/stories/stack-view/stack-view.stories.ts:102]: Was this included here intentionally? I think this should in a separate PR.

**Review by kevinbuhmann — CHANGES_REQUESTED** [review]

File: .storybook/stories/button/button-group.stories.ts [hunk 4]
```diff
@@ -26,7 +26,7 @@ export default {
   component: ClrButtonGroup,
   argTypes: {
     // inputs
-    clrMenuPosition: { control: { type: 'radio', options: CLR_MENU_POSITIONS } },
+    clrMenuPosition: { type: 'radio', options: CLR_MENU_POSITIONS },
```
Comment (kevinbuhmann) [inline .storybook/stories/button/button-group.stories.ts:29]:
```suggestion
    clrMenuPosition: { control: 'radio', options: CLR_MENU_POSITIONS },
```

Comment (kevinbuhmann) [reply .storybook/stories/button/button-group.stories.ts:29]: Please fix all of these. Replace `type: 'radio'` with `control: 'radio'`.

File: .storybook/stories/datagrid/datagrid-row.stories.ts [hunk 5]
```diff
@@ -10,6 +10,49 @@ import { moduleMetadata, StoryFn, StoryObj } from '@storybook/angular';
 
 import { elements } from '../../helpers/elements.data';
 
+export default {
```
Comment (kevinbuhmann) [inline .storybook/stories/datagrid/datagrid-row.stories.ts:13]: Moving large chunks of code in the middle of an already-large PR makes things harder to review. Please either put these back or move them in a separate PR where that is the only change.

File: .storybook/stories/dropdown/dropdown-menu.stories.ts [hunk 6]
```diff
@@ -70,6 +70,7 @@ const DropdownMenuTemplate: StoryFn = args => ({
   props: args,
 });
 
+/*
 const DropdownMenuAllTemplate: StoryFn = args => ({
```
Comment (kevinbuhmann) [inline .storybook/stories/dropdown/dropdown-menu.stories.ts:74]: Why is this story commented out? Was this a mistake?

File: .storybook/stories/forms/forms-reactive.stories.ts [hunk 7]
```diff
@@ -26,13 +26,11 @@ export default {
...
     // inputs
-    clrLabelSize: { control: { type: 'number', min: 1, max: 12 } },
+    clrLabelSize: { type: 'number', min: 1, max: 12 },
     // story helpers
     patterns: { control: { disable: true }, table: { disable: true } },
     form: { control: { disable: true }, table: { disable: true }, mapping: { [formMappingKey]: getForm() } },
-    clrLayout: {
-      control: { type: 'radio', options: Object.values(ClrFormLayout).filter(value => typeof value === 'string') },
-    },
+    clrLayout: { options: Object.values(ClrFormLayout).filter(value => typeof value === 'string') },
```
Comment (kevinbuhmann) [inline .storybook/stories/forms/forms-reactive.stories.ts:33]:
```suggestion
    clrLayout: { control: 'radio', options: Object.values(ClrFormLayout).filter(value => typeof value === 'string') },
```

File: .storybook/stories/forms/forms-reactive.stories.ts [hunk 8]
```diff
@@ -26,13 +26,11 @@ export default {
   ],
   argTypes: {
     // inputs
-    clrLabelSize: { control: { type: 'number', min: 1, max: 12 } },
+    clrLabelSize: { type: 'number', min: 1, max: 12 },
```
Comment (kevinbuhmann) [inline .storybook/stories/forms/forms-reactive.stories.ts:29]:
```suggestion
    clrLabelSize: { control: { type: 'number', min: 1, max: 12 } },
```

This is wrong. Please revert.

File: .storybook/stories/forms/forms-required-mark.stories.ts [hunk 9]
```diff
@@ -26,12 +26,13 @@ export default {
   ],
   argTypes: {
     // inputs
-    clrLabelSize: { control: { type: 'number', min: 1, max: 12 } },
+    clrLabelSize: { type: 'number', min: 1, max: 12 },
```
Comment (kevinbuhmann) [inline .storybook/stories/forms/forms-required-mark.stories.ts:29]:
```suggestion
    clrLabelSize: { control: { type: 'number', min: 1, max: 12 } },
```

This is wrong. Please revert.

File: angular.json [hunk 10]
```diff
@@ -101,6 +101,38 @@
       "sourceRoot": "projects/angular/src",
       "prefix": "lib",
       "architect": {
+        "storybook": {
```
Comment (kevinbuhmann) [inline angular.json:104]: nit: I think these should be after `build` and `test`.

Comment (kevinbuhmann) [reply angular.json:104]: And I would rename this one to `start-storybook` for clarity.

**Review by kevinbuhmann — CHANGES_REQUESTED** [review]

File: playwright.config.ts [hunk 11]
```diff
@@ -21,7 +21,7 @@ export default defineConfig({
   snapshotPathTemplate: './tests/snapshots/{arg}{ext}',
   timeout: 30 * 1000,
   expect: {
-    timeout: 5000,
+    timeout: 8000,
```
Comment (kevinbuhmann) [inline playwright.config.ts:24]: This seems long. Are we sure we need to do this?

File: projects/angular/src/utils/cdk/cdk-trap-focus.module.ts [hunk 12]
```diff
@@ -4,16 +4,21 @@
...
 
 /**
  * This is just a copy of CdkTrapFocus so it can be used independent of the rest of the A11yModule.
  */
 @Directive({
   selector: '[cdkTrapFocus]',
 })
-export class CdkTrapFocusModule_CdkTrapFocus extends CdkTrapFocus {}
+export class CdkTrapFocusModule_CdkTrapFocus extends CdkTrapFocus {
+  constructor(elementRef: ElementRef, focusTrapFactory: FocusTrapFactory, @Optional() @Inject(DOCUMENT) document: any) {
```
Comment (kevinbuhmann) [inline projects/angular/src/utils/cdk/cdk-trap-focus.module.ts:18]: Please add the same comment as the drag module here and in the trap focus module.

File: projects/angular/src/utils/cdk/cdk-drag.module.ts [hunk 13]
```diff
@@ -4,16 +4,47 @@
...
   selector: '[cdkDrag]',
 })
-export class CdkDragModule_CdkDrag extends CdkDrag {}
+export class CdkDragModule_CdkDrag extends CdkDrag {
+  /**
+   * Include the constructor to forward all the dependencies to the base class
+   * as a workaround to fix Angular "ɵɵinvalidFactoryDep" error after upgrading storybook
+   * https://github.com/storybookjs/storybook/issues/23534
+   */
+
```
Comment (kevinbuhmann) [inline projects/angular/src/utils/cdk/cdk-drag.module.ts:33]: nit: unnecessary blank line

File: .storybook/preview.js [hunk 14]
```diff
@@ -8,23 +8,15 @@ import '@cds/core/icon/register.js';
...
-import clrUiStyles from 'raw-loader!sass-loader!../projects/ui/src/clr-ui.scss';
-
 import { THEMES } from './helpers/constants';
 
 const privateModifier = 121;
 const cdsThemeAttribute = 'cds-theme';
-const styleElement = addStyleElement();
-
-const cdsCoreStyles = [previewStyles, resetStyles, coreTokens, darkThemeCoreStyles, clrUiStyles];
+addStyleElement();
```
Comment (kevinbuhmann) [inline .storybook/preview.js:19]: This should be removed because it's not needed anymore. (Please remember to remove the function declaration as well.)

**Review by kevinbuhmann — CHANGES_REQUESTED** [review]

File: .storybook/stories/signpost/signpost.stories.ts (hunk 1 above)
Comment (andyfeds) [reply .storybook/stories/signpost/signpost.stories.ts]: Reverted to the previous code.

File: .storybook/stories/button/button-group.stories.ts (hunk 4 above)
Comment (andyfeds) [reply .storybook/stories/button/button-group.stories.ts]: Fixed the above.

File: .storybook/stories/forms/forms-reactive.stories.ts (hunk 7 above)
Comment (andyfeds) [reply .storybook/stories/forms/forms-reactive.stories.ts]: Fixed this.

File: angular.json (hunk 10 above)
Comment (andyfeds) [reply angular.json]: Renamed to `start-storybook` and moved it below `build` & `test`

File: package.json [hunk 15]
```diff
@@ -53,7 +53,7 @@
     "_lint:styles:fix": "npm run --silent _lint:styles -- --fix",
     "_lint:styles:changed": "node ./scripts/get-changed-files.js --command \"stylelint\" --filter \"**/*.{css,scss,sass}\"",
     "_lint:styles:changed:fix": "node ./scripts/get-changed-files.js --command \"stylelint --fix\" --filter \"**/*.{css,scss,sass}\"",
-    "_start:storybook": "wait-on ./documentation.json && start-storybook -p 6006"
+    "_start:storybook": "wait-on ./documentation.json && ng run clr-angular:storybook"
```
[the remark this answers is not in the document]
Comment (andyfeds) [reply package.json]: Removed compodoc specific scripts

File: playwright.config.ts (hunk 11 above)
Comment (andyfeds) [reply playwright.config.ts]: Reverted this, had changed it to test an intermittent issue with the visual tests.

File: .storybook/stories/stack-view/stack-view.stories.ts (hunk 3 above)
Comment (andyfeds) [reply .storybook/stories/stack-view/stack-view.stories.ts]: This was added in this PR by mistake, It is part of the CDK popover refactoring, it will be raised along with those changes. Reverted the same in this PR.

File: .storybook/stories/dropdown/dropdown-menu.stories.ts (hunk 6 above)
Comment (andyfeds) [reply .storybook/stories/dropdown/dropdown-menu.stories.ts]: Playwright was causing intermittent issues with full page screenshots for this story, reverted the change.

File: .storybook/stories/datagrid/datagrid-row.stories.ts (hunk 5 above)
Comment (andyfeds) [reply .storybook/stories/datagrid/datagrid-row.stories.ts]: Reverted this change.

**Review by kevinbuhmann — APPROVED** [review]
